File: coding/gfg/DSA/python/binarysearch.py
def binary_search(l, x, low, high):
  if high >= low:
    mid = (high + low) // 2
    if l[mid] == x:
      return mid
    elif l[mid] < x:
      return binary_search(l, x, 0, mid)
    else:
      return binary_search(l, x, mid + 1, high)
  else:
    return -1

l = [int(i) for i in input().split()]
x = int(input())
print(binary_search(l, x, 0, len(l) - 1))

File: code/coding/gfg/DSA/cpp/binarysearch.cpp
#include <iostream>
#include <vector>

using namespace std;

int binary_search(int n, int x, vector<int> v, int low, int high) {
  int mid = (low + (high - low)) / 2; 
  if (v[mid] == x) {
    return mid;
  }
  if (v[mid] > x) {
    return binary_search(n, x, v, mid + 1, high); 
  } else {
    return binary_search(n, x, v, low, mid - 1);
  }
}

int main() {
  int n, x;
  cin >> n >> x;
  vector<int> v(n);
  for (auto &i : v) {
    cin >> i;
  }
  int low = 0, high = v.size();
  int pos = binary_search(n, x, v, low, high);
  cout << "pos : " << pos << "\n";
  return 0;
}